### PR TITLE
Add project structure

### PR DESCRIPTION
**Pull request by denscholar**
## This PR is for 'Project-structure milestone' :

- Structured the application files using the "feature folder" approach and the use the ducks pattern.
- The building blocks of the app is set as re-usable components.
- Added two routes for the app's navigation namely:
1. Books - the default view
2. Categories - that displays  "Under construction" 

- Created a component called Book for displaying a single book.
- Created a form component for adding a book (no functionality yet)

### CODE REVIEW

File: src/components/Books/Books.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { Container } from 'react-bootstrap';
+import './Books.css';
+import Book from '../Book/Book';
+import Form from '../Form/Form';
+
+const Books = () => (
```
Comment (Meltrust) [inline src/components/Books/Books.js]: According to the project requirements:

image

Therefore:

- [ ] Please, kindly make your "Book" take the data as props, so it's actually reusable in a React way and it's ready for the data.  "Book" will be responsible for rendering each book with the title and the author taken as props.

Don't worry, I'll help you.

## Explanation

We need to create a reusable component that has each book's data.

## Hints

On your "Book.js" component and make it take the book data as a prop.  When done, include it inside your "Books" component as a child.

Look at this example:

### In Book.js:

```JAVASCRIPT

// Eslint won't let you add props without these prop types validations.  
// In order to use them, you need to install them: npm install prop-types --save
import PropTypes from 'prop-types';  


// The Book component.  Takes the data as props to render the title and author.
const Book = ({ title, author }) => (  //These are the props!

      <div>
        <p>{title}</p>// The title prop
        <button type="button">
          {author} // The author prop
        </button>
      </div>
}

// After the function component, add validations to satisfy eslint.

Book.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
};

export default Book;
```

### On your Books.js:

```JAVASCRIPT

import Book from './Book';

const Books = () => {

  const booksList = [ // This is an example list of initial books.
    { id: 1, title: 'The Hobbit', author: 'J.R.R. Tolkien' },
    { id: 2, title: 'The Da Vinci Code', author: 'Dan Brown' },
    { id: 3, title: 'Pinocchio', author: 'Carlo Collodi' },
  ];

  return (
    <>

      <ul>
        {booksList.map((book) => ( // Iterate over the list of initial books.  
          <li key={book.id}>
            <Book title={book.title} author={book.author} /> // Here we are using the imported Book component and feeding it with the props data!!
          </li>
        ))}
      </ul>

    </>
  );
};

export default Books;
```

Now that you know how to do it, please make your "Book" component accept props, and include it in your "Books" component.  That way, we will meet this requirement, and the components will be ready for the data. 💪  


_Note: This is just an example to help you, and should not be taken as an "official way" to do it. This is just the result of my own personal effort and experience._

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @denscholar,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- Nice styles✔️
- Nice readme!! ✔️ 
- Nice folder structure ✔️  

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Books/Books.js (hunk 1 above)
Comment (denscholar) [reply src/components/Books/Books.js]: @Meltrust Thank you for your feedback. However, you failed to recognize that there are many ways to render components. From my project implementation, I rendered the Book component and it's already waiting to receive data which will eventually be passed to the Book.js file as props from the Books.js. I must not have dummy data before I can render the component. Anyway, I already fixed it based on your recommendation. Thank you.

Comment (Meltrust) [reply src/components/Books/Books.js]: Hey @denscholar, 

The **_only_** request that I left was to make your `Book` component to take the book data as props:

image

The other content that you are referring to was in different help sections clearly labeled as "Explanation" and "HInts", which I wrote with **_examples_** in order to help you.   Both mentioned sections start after the phrase: "don't worry, I'll help".  

I'm glad this helped you in some way.

Thank you for your kind feedback.

**Review by rustam-tolipov — APPROVED** [review]
## Hello @denscholar 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
## Congratulations! 🎉 🎉 🎉 🎉 🎉, 

Cheers and Happy coding!👏👏👏

Developer-512px-10

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**